src/main/java/com/restapi/carMarket/service/CarService.java

```diff
@@ -20,68 +20,96 @@
     @Autowired
     CarDao carDao;
 
-    public void insert(Car car) {
-        if(isValid(car))
-            carDao.save(car);
-        else
-            throw new CarNotValidException();
-    }
-
-    private boolean isValid(Car car) {
-        boolean valid = true;
-        if(isValidPrice(car))
-            valid = false;
-        else if(isValidYear(car))
-            valid = false;
-        return valid;
-    }
-
-    private boolean isValidYear(Car car) {
-        return car.getYear()<1885 || car.getYear()> LocalDateTime.now().getYear();
-    }
-
-    private boolean isValidPrice(Car car) {
-        return car.getPrice()<1 || car.getPrice()>2000000000;
-    }
-
-    public List<Car> findAll() {
+    public List<Car> getAllCarsOnMarket() {
         return carDao.findAll();
     }
 
-    public Car findById(Long id) {
+    public Car getCarById(Long id) {
         Optional<Car> car = carDao.findById(id);
 
         if(car.isPresent())
             return car.get();
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("Car with id("+id+") is not found");
     }
 
-    public void delete(Car car) {
+    public void removeCarFromMarket(Car car) {
         if(carDao.exists(Example.of(car)))
             carDao.delete(car);
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("A Car with properties("+car.toString()+") is not found");
 
     }
 
-    public void deleteById(Long id) {
+    public void removeCarFromMarketById(Long id) {
         if(carDao.existsById(id))
             carDao.deleteById(id);
```

Comment:
Interesting that you have to call the dao to check car existence... What will happen, if you simply call `carDao.deleteById(id)` and that ID does not exist? Will the dao throw an exception? Maybe you could just simply catch that exception and translate it to CarNotFoundException.
Another option to avoid duplicating "car not found" logic is to reuse getCarById() method here:
```
Car car = getCarById(id);
carDao.delete(car);
```
```

Reply:
I will use getCarById as you wrote. Thanks!

Reply:
done